lib/omniauth/strategies/saml/auth_request.rb

```diff
@@ -35,7 +35,14 @@ def create(settings, params = {})
           end
 
           encoded_request   = CGI.escape(base64_request)
-          request_params    = "?SAMLRequest=" + encoded_request
+          delimiter =
+            if settings[:idp_sso_target_url].include?('?')
+              '&'
+            else
+              '?'
+            end
+
+          request_params = "#{delimiter}SAMLRequest=#{encoded_request}"
 
```

Comment:
This seems like a lotta lines to do a & or ?. Don't like the inline version ?

delimiter = settings[:idp_sso_target_url].include?('?') ? '&' : '?'

?